packages/backend/src/open_payments/payment_pointer/service.ts

```diff
@@ -127,12 +135,28 @@ async function updatePaymentPointer(
 ): Promise<PaymentPointer | PaymentPointerError> {
   try {
     const update: UpdateInput = { publicName }
-    if (status) {
-      update.deactivatedAt = status === 'INACTIVE' ? new Date() : null
+    const paymentPointer = await PaymentPointer.query(deps.knex)
+      .findById(id)
+      .throwIfNotFound()
+
+    if (status === 'INACTIVE' && paymentPointer.isActive) {
+      update.deactivatedAt = new Date()
+      const expiresAt = new Date(Date.now() + deps.config.gracePeriodMs)
+
+      await IncomingPayment.query(deps.knex)
+        .patch({ expiresAt })
+        .where('paymentPointerId', id)
+        .whereIn('state', [
+          IncomingPaymentState.Pending,
+          IncomingPaymentState.Processing
+        ])
```

Comment:
I figured these incoming payment states constitute "open" payments based off the names/comments in the enum. Is that accurate?

Reply:
This is accurate:
`IncomingPaymentState.Processing` -> payment is currently being streamed into. Typically doesn't happen in our P2P scenarios, since we just receive a 
`IncomingPaymentState.Pending` -> payment was created, no $ has streamed into it yet.

Reply:
@sabineschaller Could you chime in here? My understanding is that we want to push the `expiresAt` back so that we ensure the payment has at least that amount of time to complete. 

I think this turns into a no-op if we filter out the existing `expiresAt` that are less than the new `expiresAt`. From what I can see the max expiration date for incoming payments created via openapi is the creation time + 1 day. https://github.com/interledger/rafiki/blob/2a3242a355551044230e02597cbf9c98cf211347/packages/backend/src/open_payments/payment/incoming/routes.ts#L32

So if we only get the incoming payments with an `expiresAt` that is less than now + 1 day, we will never have incoming payments because they have at max some past date + 1 day.

Reply:
Just to confirm from Lisbon: we wanted 90 days in the future as default for expiryDate for incoming payments, correct?

Reply:
I think we actually settled on 30 days.